crates/ruff_cli/src/lib.rs

```diff
@@ -141,21 +153,31 @@ pub fn run(
             commands::version::version(output_format)?;
             Ok(ExitStatus::Success)
         }
-        Command::Rule { rule, all, format } => {
+        Command::Rule {
+            rule,
+            all,
+            format,
+            mut output_format,
+        } => {
+            output_format = warn_about_deprecated_help_format(output_format, format);
             if all {
-                commands::rule::rules(format)?;
+                commands::rule::rules(output_format)?;
             }
             if let Some(rule) = rule {
-                commands::rule::rule(rule, format)?;
+                commands::rule::rule(rule, output_format)?;
             }
             Ok(ExitStatus::Success)
         }
         Command::Config { option } => {
             commands::config::config(option.as_deref())?;
             Ok(ExitStatus::Success)
         }
-        Command::Linter { format } => {
-            commands::linter::linter(format)?;
+        Command::Linter {
+            format,
+            mut output_format,
+        } => {
+            output_format = warn_about_deprecated_help_format(output_format, format);
```

Comment:
Perhaps `resolve_help_output_format` would be a better function name? No strong feelings though.